tests/worker/ingestion/action-manager.test.ts

```diff
@@ -0,0 +1,85 @@
+import { PluginsServer, PropertyOperator } from '../../../src/types'
+import { createServer } from '../../../src/utils/db/server'
+import { createRedis, delay } from '../../../src/utils/utils'
+import { ActionManager } from '../../../src/worker/ingestion/action-manager'
+import { resetTestDatabase } from '../../helpers/sql'
+
+describe('ActionManager()', () => {
+    let server: PluginsServer
+    let closeServer: () => Promise<void>
+    let actionManager: ActionManager
+
+    beforeEach(async () => {
+        ;[server, closeServer] = await createServer()
+        await resetTestDatabase()
+        actionManager = new ActionManager(server.db)
+        await actionManager.prepare()
+    })
+    afterEach(async () => {
+        await closeServer()
+    })
+
+    describe('getAction()', () => {
+        it('returns the correct action', async () => {
+            const action = actionManager.getAction(67)
+
+            expect(action).toMatchObject({
+                id: 67,
+                name: 'Test Action',
+                deleted: false,
+                post_to_slack: false,
+                slack_message_format: '',
+                is_calculating: false,
+                steps: [
+                    {
+                        id: 911,
+                        action_id: 67,
+                        tag_name: null,
+                        text: null,
+                        href: null,
+                        selector: null,
+                        url: null,
+                        url_matching: null,
+                        name: null,
+                        event: null,
+                        properties: [{ type: 'event', operator: PropertyOperator.Exact, key: 'foo', value: ['bar'] }],
+                    },
+                ],
+            })
+
+            await server.db.postgresQuery(
+                `UPDATE posthog_actionstep SET properties = jsonb_set(properties, '{0,key}', '"baz"') WHERE id = 911`,
+                undefined,
+                'testKey'
+            )
+
+            // This is normally done by Django async in such a situation
+            await actionManager.fetchAction(67)
```

Comment:
Yup, echoing my point from above, the name is misleading here. What happens to the fetched action? Not obvious it actually caches it in a map... :)

Reply:
Other than calling this here, should we also have a bigger and more E2E test to make sure actions actually reload in the workers? Fine to push that for a later PR.

Reply:
At this stage it's hard to test this E2E with the pubsub only in the main thread (whole `startPluginsServer` needed instead of just `createServer`), I don't know about reaching into `actionManagers` in worker threads, and there's no actual functionality (like `onAction`) to test here yet. But definitely a goal to have E2E tests when more parts are in place.